go/merkletree2/tree.go

```diff
@@ -156,17 +170,36 @@ type PositionHashPair struct {
 
 type RootMetadata struct {
 	// TODO Add timestamp, version.....
-	_struct          struct{} `codec:",toarray"` //nolint
-	Seqno            Seqno    `codec:"n"`
-	BareRootHash     Hash     `codec:"r"`
-	SkipPointersHash Hash     `codec:"s"`
+	_struct          struct{}     `codec:",toarray"` //nolint
```

Comment:
version?

Reply:
The EncodingType is kind of a version itself, we can always bump to signal other sorts of updates. Do not feel strongly about it though, if you think another version field makes it more clear/explicit I am happy to add it. 
I should also actually add code to check that the EncodingType is as expected in the proofs so that clients can throw a special error and be asked to update (rather than have proofs generically fail). I should also remove that TODO. Do we need a timestamp?